### DISCUSSION TITLE
Credit based economy on the Station

### DISCUSSION THREAD

**Discussion by BorjaUnizar** [space-wizards/space-station-14#21816] [Ideaguys]
## Description

This idea is a bit long but I would appreciate it if you could read it completely and give some feedback. Given the depth of it I don't think it could be implemented in the near future but I've been thinking about it for a while and I wanted to share it nonetheless.

Like the title suggests, it would be implementing a credit system similar to a combination of SS13's Goonstation and SS14's Frontier Station, with some tweaks to encourage more RP interactions and not just add a new obstacle for crewmates who want to drink a can of space cola.

**How would the system work?**

- Give the station a General budget. This budget would behave just like nowadays' cargo, having a natural tick (credits per second) growth and being increased via cargo's sales too. This budget would be divided into two subgroups: The Payroll budget and the Supply budget.

- The Payroll budget would be used, well, to pay the crew. Every crew member would have a certain salary depending on their job, which could be increased/decreased depending on their performance or other events.

- The Supply budget would be used almost just like it already is, to pay for materials and other items needed by the different departments, with the added difference that given that crew members now have their own money too, they could make personal requests to cargo, which would be a new way for them to increase their budget too (*summons interest rate demon*).
 
- The Head of Personnel would manage the General budget, deciding what portion of it is allocated to the two different subgroups, and the Payroll budget, being responsible for the crew members' salaries. The Quartermaster, obviously, would manage the Supply budget.

- The credit account of every crew member would be linked to their ID. The ID could be used to retrieve creditbucks at an ATM (which would need to be added) or at the HoP's office (using the ID console or a new machine would need to be added). Payrolls would be automatically added to everyone's account at the end of the shift, or manually by the HoP if a crew member needs to leave the round early, after notifying them. The amount would be stored for future rounds.

**What would the credits be used for?**

Let's take out the obvious one first, yes, vending machines. The credits used this way would flow back to the General budget, or even better, they could stay inside the vending machine.

- Besides that, as previously stated, crew members would be able to make personal requests to cargo, which wouldn't require any head's approval (as long as you're not ordering laser guns or other similar *"funny"* requests) . On top of that, cargo should never refuse these, as not only they can't lose money, but if they're smart enough they will win some. Also, uh, it's their job.

- Given that salaries in the Service department would typically be lower than those of other departments, these roles could charge credits for their products. For example, Chefs, Bartenders, or Botanists could have different prices for their offered items, and Clowns or Mimes could charge for their... *"performances"*. This way higher paid roles would use their salary to consume the products made by the lowered paid Service roles, evening it out.

- Looking at it from the antag(or not so antag) point of view, credits could be used in many malicious ways too. Need someone to do some risky tasks for you? Offer some credits. Want to bribe someone for something you shouldn't have? Nothing a couple of spacebucks can't do. "Whoops, your PDA fell into my thieving gloves, time to order 25 gorilla crates". The options are limitless. 

**What is the desired effect of the system?**

This may be more of a personal opinion but on MRP/HRP Servers the budget, compared to the station's power,structure,security,etc... is usually not a problem the crew has to deal with. With this change I wouldn't want to overshadow the rest of events that unfold on every unique shift of this great game, but add a new layer of complexity and most importantly, new RP opportunities. 

Power struggles between Command and Cargo over the general budget, crewmates asking the HoP for a permit to start their own shop, Command asking CentComm for a higher budget to fullfill the stations needs and all of the mentioned ideas above are examples of the new possibilities it would bring.

**The ugly but last part, the numbers**

If you're actually still reading, I'm amazed, thank you. This is the only part i haven't thought about in depth. To be fair I'd just try to follow a similar line to Frontier Station for the numbers, and tweak it a bit if needed, but that would need to have all of this implemented first, and this is just me presenting an idea, so I won't delve more into it.

**Comment by EthanQix** [comment]
I'm not sure what this system would actually *add* in terms of interesting mechanics or RP opportunities, apart from the tider shops and overall goods smuggling. Or even if it would be worth it.

You would have to pay for a beer at the bar, or a meal at kitchen, or fresh fruit at botany. You'd have to pay to watch the clown's performance at the theater. Would you have to pay for getting healed at medbay ? Would science have to pay for materials from cargo ? Would players have to pay for science gadgets ? Would departments have to pay for repairs after an errant meteor ?

The problem with an economy system is that it is ubiquitous, affecting all parts of the station and all jobs. It is also horrendously tedious to balance - I'm talking NewMed level of tedium here. This would require a design doc, starting with a more robust list of positive effects than "I think it would bring some interesting RP opportunities". Good luck.

**Reply by Everturning** [reply]
This, but also: there IS a certain emptiness that comes when you are doing a tider shop. There's no transaction that occurs, no passing of goods. Right now it's just. "want insuls" "yes" "okay" and then the encounter is over. There's no way to justify making a shop without currency gain, because of that emptiness. Sure, you can trade, but that's somewhat hard to do.

**Reply by EthanQix** [reply]
Well bartering and haggling are a form of RP ^^

**Reply by superjj18** [reply]
gambling, bartering, bragging rights at the round end screen to see who made the most money in the shift, a reason to break into the vault, limiting the amount of things people can take from a vending machine.  plenty of things this could add.

**Comment by forgotmyotheraccount** [comment]
I remember that vgstation had a bookkeeping PR. The librarian would have their own console, and solve math equations to provide small boosts to the station payroll. Not sure if it ever got pushed in the end but either way it'd be a fun to interact with the payroll if salaries were ever implemented.

**Comment by ArcaneOcto** [comment]
I am 100% ON BOARD with implementing a Frontier-ish economy to Wizden servers. If this is done correctly I think it could be one of the coolest features added to the game, however that is a big "if". Here's a few of my thoughts, both on features and balancing:

+ I think, do to the fact that Upstream servers are much more compact and action-packed, that any economy system added needs be as streamlined and painless as possible. Having to say, worry about paying the medical staff, could slow the game down a TON. To avoid this I think that some features should be automated such as getting paid after the round ends like you proposed. Medical fees could be done this way... The game could keep track how many times a doctor revived you, then charge you a small fee for each instance at the end of the game. The money collected here could serve as a "tip" of sorts to the medical department. This would encourage Medical to revive as many people as possible (more money that way) but not slow down the game whatsoever.

+ The concept of your entire bank account being accessible via your ID is cool, but I think could get frustrating in practice. Instead, I propose that all purchases are made with physical stacks of Spesos... which can then be withdrawn/deposited at ATMs. The ATMs would only be in one or two places on the station ensure that players still have (stealable) money on them (nobody wants to constantly walk back to the ATM spot every time they want to buy something). This means that players would be carrying around plenty of money to buy things with, but wouldn't go completely bankrupt if an antagonist decided to mug them. Does that make sense? The progress represented buy a bank account is real to the player, and it would SUCK to lose it all to a Syndie who thought buying 20 gorilla crates would be funny. They could still do that, but it would be your fault... because you were stupid (or rich) enough and carried around way too much pocket change.

+ This could potentially put a lot of strain on the HoP role, and possibly make some people upset. I think most people agree that managing money is _not_ fun. I think that having a budget split between cargo and crew paychecks is a fantastic idea... _but_ I think that it needs to be implemented in a way that is quick, fun, and easy for the HoP to do. My addition to your concept here is this: the HOP has a special console that can be used for two things: First is adjust how the station's money is split using just two buttons. It would show what percentage the crew is getting paid on the left, and what percentage cargo gets on the right. Clicking the either button tilts the ratio to the corresponding side. For example, say that the money is being split 50/50 initially, the HoP could click the right button a few times to make it split 15% to 85%. This could be done quickly and in a pinch with **no math** (the HoP could quickly change it based on demand and get back to their other duties). This ratio should only effect how new income is split up, and leave the current balances unchanged. This is in order to prevent the HoP from just allocating all of the funds the Crew right before going to Evac. The second thing this console should be able to do is select one department to get a bonus at the end of the shift. This adds a fun reward for the HoP to bestow, and also adds more motivation for the departments to do their best (and it also adds a bit of healthy competition).

+ In order to keep things from getting stale (eg. everyone having a million plus spesos after hours of gameplay) there has to be something preventing literally every player from becoming filthy rich after a handful of rounds. This could potentially be fixed by implementing the medical bills as I mentioned, which would also serve to reward people skilled at being alive and to add more fear to dying in game. It could also be fixed by capping bank accounts or wiping them every so often (kinda lame imo).

I also thought I'd just list a few concepts why an economy system would be so cool:

+ There could be money-based events. For example a small shuttle could show up that is all beat up. Repairing it according to a blueprint would grant Engineering a bonus. Or maybe there is a cache hidden somewhere on the station (I think there's already a PR for this). Or maybe a hyper secure vault mysteriously FTLs into the region for Cargo/Salv to go retrieve and *definitely* share with the rest of the crew ;)

+ SO many interesting things to do on the side of EVIL. Objectives about stealing money (off people or maybe places like the Bar could have a small safe). Bribes. MONEY PRINTING should be a thing for the more tech-savey individuals. Drug dealing. Poker. Stealing cargo's haul of money before they can deposit it in the ATM.

+ More things to do as a Security guard (or a tider turned bodyguard). Helping the barkeep defend his safe. Definitely not keeping money found on criminals ;). Providing an escort to cargo when they go to deposit money in the ATM. Arresting engineers trying to print money in maints.

+ MORE THINGS TO DO. Open random stores and get paid. Accept a dare to eat a maints pill and get paid. Win a round of boxing and get paid. Lose a bet and go broke. Pay the clown to stop slipping the captain (they keep doing it anyway). Buy a round of drinks for your department. Absolutely do not start printing money in maints ;). Pay the travelling chef for that exotic sushi that absolutely won't kill you. Hire a tider bodyguard. Pay people to let you interview them. Maybe the reporter could get paid whenever someone reads their articles or tunes into the broadcast.

ANYWAY. My bad for the massive wall of text, but hopefully some of this is useful feedback as well as shows why this is _really really cool and awesome mechanic that needs to be added_

**Reply by Aisu9** [reply]
> * In order to keep things from getting stale (eg. everyone having a million plus spesos after hours of gameplay) there has to be something preventing literally every player from becoming filthy rich after a handful of rounds. This could potentially be fixed by implementing the medical bills as I mentioned, which would also serve to reward people skilled at being alive and to add more fear to dying in game. It could also be fixed by capping bank accounts or wiping them every so often (kinda lame imo).

I do not quite agree on this front, having a permanent economy is an invitation for disaster(it is an horrible additional maintenance) AND giving old players an advantage and new players a disadvantage, if an economy will be applied it should be a round-to-round basis where everything reset like on TG or Goon if you played there

**Reply by fidso** [reply]
I think the part of accessible via ID is actually good, if you think about it. If money is easily spent, then this would balance out the "players getting filthy rich after some rounds" problem. Also, people would value money because it saves between rounds, but it wouldnt turn in super ultra filthy rich players controlling everything, cause big chances you would get robbed sometime.

**Reply by ArcaneOcto** [reply]
I'm not sure if an economy system is worth adding of it isn't at least partially permanent. There are ways to make it so that old players don't have as much of an advantage (wipes every month, limit how much money you can bring to a shift based on your role, etc). If the money you earn does not stick around, then a lot of that sense of progression you get from earning cash goes away. Money in this case would have all of the annoying downsides and none of the upsides that emerge from having the money mean something long-term. It would essentially just end up an rp tool and a limit on how much you can use vendong machines. Maybe that would be ok, but I'd much prefer a deeper and more impactful mechanic.

**Reply by Aisu9** [reply]
considering 100% of the current progressions available ot us are DURING rounds, i'm adamant adding an OUTSIDE progression is a very bad thing, even with limitation per role, it's additional maintenance effort, monthly wipe? additional maintenance, there is no long-term benefit so far and none should be considered as it's a waste of efforts on maintaining this system.
It would push people to work a bit to get better or additional things, design problem so far is bounties are cargo only, and we would greatly gain from moving bounty notes ot bounty cubes with a public bounty console available to anyone, so everyone can produce wealth for themselve and cargo(and by extension the overall budget to pay everyone on pay roll)

**Reply by ArcaneOcto** [reply]
> I think the part of accessible via ID is actually good, if you think about it. If money is easily spent, then this would balance out the "players getting filthy rich after some rounds" problem. Also, people would value money because it saves between rounds, but it wouldnt turn in super ultra filthy rich players controlling everything, cause big chances you would get robbed sometime.

This is a good point. However I really like the idea of the ATMs and buying things with physical spesos. What if you still needed to buy things with cash, but you could steal an ID and get cash from the ATM using the owners account? That way, you could still drain CE's bank account... But you'd need to get to an ATM with his ID (without getting caught) to actually do so. Another concern is that if you use your ID like a debit card like was originally suggested, then we need a whole bunch of new consoles to buy stuff from. For example, how do you pay the chef with just your ID? You'd need a new console. It would be a lot easier just to pass them some spesos.

**Reply by Aisu9** [reply]
well i didn't say the overall concept of having a pay is bad, it add a degree of dynamic to each rounds because people when they have free time will take bounties and make money this way, and use this money on vends or in cargo, there is possibility to purchase things with cargo's budget but you can also pay "privately" using your own account.
On TG you can straight up withdraw credits from the ID, on Goon you need to use ATM. when using an ATM might be good, people will most likely get pissy and request or do the change themselve because limited ATMs means traffic jam, also probably another unique machine that wont be indestructible and not buildable, what will happen if a rod destroy both? well the entire station wont be able to use their money anymore.

**Comment by ArcaneOcto** [comment]
Also, this is a year old? Why is this just now getting talked about

**Comment by Gigawatsup** [comment]
After expirencing Monkestations credit system on ss13, your account is reset every round but there are plenty of ways to make money and lose it to such as in vending machines and to service who benefit greatly from getting credits since they don't get paid well with the HOP managing payroll and the QM managing station funds being restricted from just giving cargo all of the money
